tests/unit/models/test_utils.py

```diff
@@ -0,0 +1,206 @@
+) -> None:
+    write_summary_kernel_parameters(kernel)
+
+    assert mocked_summary_scalar.call_count == len(names)
+    for i, (n, v) in enumerate(zip(names, values)):
+        assert mocked_summary_scalar.call_args_list[i][0][0] == n
+        assert mocked_summary_scalar.call_args_list[i][0][1].numpy() == v
+
+
+@unittest.mock.patch("trieste.models.gpflow.interface.tf.summary.scalar")
+@pytest.mark.parametrize(
+    "likelihood, names, values",
+    [
+        pytest.param(
+            gpflow.likelihoods.Gaussian(),
+            ["likelihood.Gaussian.variance"],
+            [1],
+            id="Gaussian, Default",
+        ),
+        pytest.param(
+            gpflow.likelihoods.Gaussian(scale=0.2),
+            ["likelihood.Gaussian.scale"],
+            [0.2],
+            id="Gaussian, scale",
+        ),
+        pytest.param(
+            gpflow.likelihoods.Gaussian(scale=gpflow.functions.Polynomial(degree=2)),
+            ["likelihood.Gaussian.scale.w"],
+            [[1, 0, 0]],
+            id="Gaussian, polynomial",
+        ),
+        pytest.param(
+            gpflow.likelihoods.Gaussian(
+                variance=gpflow.functions.SwitchedFunction(
+                    [
+                        gpflow.functions.Constant(1.0),
+                        gpflow.functions.Constant(1.0),
+                    ]
+                )
+            ),
+            [
+                "likelihood.Gaussian.variance.functions[0].c",
+                "likelihood.Gaussian.variance.functions[1].c",
+            ],
+            [1, 1],
+            id="Gaussian, grouped noise variance",
+        ),
+        pytest.param(
+            gpflow.likelihoods.Beta(),
+            ["likelihood.Beta.scale"],
+            [1],
+            id="Beta, default",
+        ),
+        pytest.param(
+            gpflow.likelihoods.HeteroskedasticTFPConditional(
+                distribution_class=tfp.distributions.Normal,
+                scale_transform=tfp.bijectors.Exp(),
+            ),
+            [],
+            [],
+            id="HeteroskedasticTFPConditional",
+        ),
+    ],
+)
+def test_write_summary_likelihood_parameters(
+    mocked_summary_scalar: unittest.mock.MagicMock,
+    likelihood: gpflow.likelihoods.Likelihood,
+    names: list[str],
+    values: list[float],
+) -> None:
+    write_summary_likelihood_parameters(likelihood)
+
+    assert mocked_summary_scalar.call_count == len(names)
+    for i, (n, v) in enumerate(zip(names, values)):
+        assert mocked_summary_scalar.call_args_list[i][0][0] == n
+        assert tf.reduce_all(mocked_summary_scalar.call_args_list[i][0][1].numpy() == v)
+
+
+@unittest.mock.patch("trieste.logging.tf.summary.histogram")
+@unittest.mock.patch("trieste.logging.tf.summary.scalar")
+@pytest.mark.parametrize("prefix", ["", "dummy_"])
+def test_write_summary_data_based_metrics(
+    mocked_summary_scalar: unittest.mock.MagicMock,
+    mocked_summary_histogram: unittest.mock.MagicMock,
```

Comment:
@uri-granta something weird happens here with scalar and histogram mock - the order matters and it only works correctly this way, not sure how are they exactly mapped to the variables below - any suggestion how to make this more robust?

Reply:
Yes, the [parameter order matters](https://docs.python.org/3/library/unittest.mock.html#nesting-patch-decorators):

> Note that the decorators are applied from the bottom upwards. This is the standard way that Python applies decorators. The order of the created mocks passed into your test function matches this order.